### PR TITLE
Don't do normalization if not JSON

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordSessionTests.cs [hunk 1]
```diff
@@ -127,6 +127,39 @@ public void BodyNormalizationWorksWhenMatching(string body, string contentType)
...
+            {
+                recordEntry.Request.Headers.Add("Content-Type", new[] { contentType });
+            }
+
+            recordEntry.Request.Body = bodyBytes;
+            recordEntry.RequestUri = "http://localhost/";
+            recordEntry.RequestMethod = RequestMethod.Delete;
+            RecordEntry.NormalizeJsonBody(recordEntry.Request);
+
+            // Intentionally not using Assert.Equals/NotEquals to check for reference equality
```
Comment (pakrym) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordSessionTests.cs:152]: Assert.AreSame

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs [hunk 2]
```diff
@@ -126,19 +126,22 @@ private static void DeserializeBody(RequestOrResponse requestOrResponse, in Json
...
-                using var memoryStream = new MemoryStream();
-                using var writer = new Utf8JsonWriter(memoryStream, WriterOptions);
-                using var document = JsonDocument.Parse(requestOrResponse.Body);
-                document.RootElement.WriteTo(writer);
-                writer.Flush();
-                requestOrResponse.Body = memoryStream.ToArray();
-                RecordedTestSanitizer.UpdateSanitizedContentLength(requestOrResponse);
-            }
-            catch (JsonException)
+            if (requestOrResponse.TryGetContentType(out string contentType) && contentType.Contains("application/json"))
```
Comment (pakrym) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs:129]: cotains("json")? so we handle text/json

**Review by pakrym — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordSessionTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordSessionTests.cs:152]: Ah I was looking for the XUnit equivalent, but didn't see it. It's Assert.Same.

**Review by scbedd — APPROVED** [review]

**Pull request merged**